src/Workspaces/CSharp/Portable/CodeGeneration/MethodGenerator.cs

```diff
@@ -170,6 +169,11 @@ internal static class MethodGenerator
                     {
                         tokens.Add(SyntaxFactory.Token(SyntaxKind.PartialKeyword));
                     }
+
+                    if (method.ReturnsByRef)
```

Comment:
this seems wrong.  if the method is 'ref' then we should make a ref type for the return type.